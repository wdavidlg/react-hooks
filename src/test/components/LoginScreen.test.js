const { mount } = require("enzyme")
const { default: LoginScreen } = require("../../components/09-useContext/LoginScreen")
const { default: UserContext } = require("../../components/09-useContext/UserContext")


describe('Pruebas en <LoginScreen />', () => {

    const user = {
        id: 1234,
        name: 'Jorge',
        email: 'jorge@jorge.com'
    }
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen/>
        </UserContext.Provider>
    )
    
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de ejecutar el setUser con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toHaveBeenCalledWith(user);
    })
    

    
})
