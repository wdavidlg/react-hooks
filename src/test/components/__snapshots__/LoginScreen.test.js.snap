// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h2>
    LoginScreen
  </h2>
  <hr />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
